### PR TITLE
Merge 0.23 -> main

### PR DESCRIPTION
**Pull request by armanbilge**
Straightforward merge except for re-synchronizing (ember-)server now that it's crossed in 0.23.

### CODE REVIEW

File: core/shared/src/main/scala/org/http4s/package.scala [hunk 1]
```diff
@@ -84,40 +81,4 @@ package object http4s {
   /** A stream of server-sent events */
   type EventStream[F[_]] = Stream[F, ServerSentEvent]
 
-  // Syntax to enable JS-cross compilation without splitting files
-  // Nobody will actually be able to run these methods b/c they will fail to link
```
Comment (armanbilge) [inline core/shared/src/main/scala/org/http4s/package.scala:88]: I hope you don't mind, I also took this opportunity to remove these shims and any deprecated methods that needed them.

File: server/shared/src/main/scala/org/http4s/server/staticcontent/FileService.scala [hunk 2]
```diff
@@ -64,8 +63,7 @@ object FileService {
         bufferSize: Int = 50 * 1024,
         cacheStrategy: CacheStrategy[F] = NoopCacheStrategy[F],
     ): Config[F] = {
-      val pathCollector: PathCollector[F] = (f, c, r) =>
-        filesOnly(Path.fromNioPath(f.toPath()), c, r)
+      val pathCollector: PathCollector[F] = (f, c, r) => filesOnly(f, c, r)
```
Comment (danicheg) [inline server/shared/src/main/scala/org/http4s/server/staticcontent/FileService.scala:66]: golden nitpick 😆 
```suggestion
      val pathCollector: PathCollector[F] = (p, c, r) => filesOnly(p, c, r)
```

File: project/Http4sPlugin.scala [hunk 3]
```diff
@@ -123,10 +125,9 @@ object Http4sPlugin extends AutoPlugin {
         id = subproject,
         name = s"Build $subproject",
         scalas = List(scala_212),
-        javas = List("adoptium@8"),
+        javas = List(JavaSpec.temurin("17")),
```
Comment (danicheg) [inline project/Http4sPlugin.scala:128]: Nice bump 8 -> 17 :)

File: server/shared/src/main/scala/org/http4s/server/Server.scala [hunk 4]
```diff
@@ -23,7 +23,7 @@ import org.log4s.getLogger
 abstract class Server {
   private[server] val logger = getLogger
 
-  def address: ip4s.SocketAddress[ip4s.Host]
+  def address: ip4s.SocketAddress[ip4s.IpAddress]
```
Comment (danicheg) [inline server/shared/src/main/scala/org/http4s/server/Server.scala:26]: Do we track anywhere that kind of changes (some migration guide for the 1.0)?

Comment (armanbilge) [reply server/shared/src/main/scala/org/http4s/server/Server.scala:26]: The change from `java.net.InetSocketAddress` to `com.comast.ip4s.SocketAddress` is noted here:
https://github.com/http4s/http4s/blob/main/website/src/hugo/content/changelog.md#v100-m24-2020-08-07

**Review by danicheg — APPROVED** [review]

File: project/Http4sPlugin.scala (hunk 3 above)
Comment (rossabaker) [reply project/Http4sPlugin.scala:128]: That one's an "app". :smile:

**Pull request merged**